server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -478,8 +476,18 @@ private AggregationBuilder createAssigneesFacet(IssueQuery query, Map<String, Fi
     StickyFacetBuilder assigneeFacetBuilder = new StickyFacetBuilder(queryBuilder, assigneeFilters);
     BoolFilterBuilder facetFilter = assigneeFacetBuilder.getStickyFacetFilter(fieldName);
     FilterAggregationBuilder facetTopAggregation = assigneeFacetBuilder.buildTopFacetAggregation(fieldName, facetName, facetFilter, DEFAULT_FACET_SIZE);
-    if (!query.assignees().isEmpty()) {
-      facetTopAggregation = assigneeFacetBuilder.addSelectedItemsToFacet(fieldName, facetName, facetTopAggregation, query.assignees());
+
+    Collection<String> assigneesEscaped = query.assignees();
+    if (assigneesEscaped != null) {
+      assigneesEscaped = Collections2.transform(assigneesEscaped, new Function<String, String>() {
+        @Override
+        public String apply(String input) {
+          return Pattern.quote(input);
+        }
+      });
+    }
```

Comment:
OK to extract in another method, see test case with dash in name for explanation over escaping.